File: src/core/models/module.ts
export type WorkspaceMode = 'flat' | 'modular'

export interface DetectedModule {
  displayName: string
  slug: string
  prefix: string
  confidence: 'high' | 'low'
}

export interface ModuleCounters {
  BF: number
  BR: number
  AC: number
  FT: number
  TC: number
  RN: number
}

export interface ModuleConfig {
  displayName: string
  prefix: string
  counters: ModuleCounters
}

export interface SharedCounters {
  ACT: number
  GBR: number
}

export interface Registry {
  mode: WorkspaceMode
  modules: Record<string, ModuleConfig>
  shared: SharedCounters
}

File: pythonlibrary/aibrd/models/module.py
from typing import Literal
from pydantic import BaseModel

WorkspaceMode = Literal["flat", "modular"]


class DetectedModule(BaseModel):
    display_name: str
    slug: str
    prefix: str
    confidence: Literal["high", "low"]


class ModuleCounters(BaseModel):
    BF: int = 0
    BR: int = 0
    AC: int = 0
    FT: int = 0
    TC: int = 0
    RN: int = 0


class ModuleConfig(BaseModel):
    display_name: str
    prefix: str
    counters: ModuleCounters = ModuleCounters()


class SharedCounters(BaseModel):
    ACT: int = 0
    GBR: int = 0


class Registry(BaseModel):
    mode: WorkspaceMode = "flat"
    modules: dict[str, ModuleConfig] = {}
    shared: SharedCounters = SharedCounters()
